mangoPURE/models/base.py

```diff
@@ -0,0 +1,63 @@
+import torch
+from ..data.base import TimedAudioBatch, MixedExample
+from typing import Any
+from dataclasses import dataclass
+
+
+class Embedder(torch.nn.Module):
+    def forward(self, batch: TimedAudioBatch) -> TimedAudioBatch:
```

Comment:
Why does it return TimedAudioBatch and not torch.Tensor?

Reply:
Because the embedder should pass the information about the noise/speaker segments